ldap/servers/plugins/replication/repl_extop.c

```diff
@@ -1175,6 +1175,10 @@ multimaster_extop_EndNSDS50ReplicationRequest(Slapi_PBlock *pb)
                 */
                 if (replica_is_flag_set(r, REPLICA_LOG_CHANGES) && cldb_is_open(r)) {
                     replica_log_ruv_elements(r);
+                    /* now that the changelog is open and started, we can alos cretae the
+                     * keep alive entry without risk that db and cl will not match
+                     */
+                    replica_subentry_check(replica_get_root(r), replica_get_rid(r));
```

Comment:
When is this call normally made? What's the problem that this resolves? It would be good to understand what this fix is changing.

Reply:
I will try to clarify what lkrispenz and tbordaz explained in the issue:
  When the keep alive entry feature was originally added, tbordaz added a call to replica_subentry_check
    in replica_enable_replication to insure that keep alive entry is already there.
 (Another place it is called is on supplier side when initializing a replica agremeent)

In ticket [48445](https://pagure.io/389-ds-base/issue/48445) the creation of a keep alive entry after online init was 
 removed because of race condition with the changelog handling but ludwig
And some time later this issue was created because keep alive entry is not created on the master (that are init by another master) and lkrispenz proposed to add it when ending the online init session once changelog is fully started

If your question is what keep alive entries are for and what is their impact, then here is the answer:
  As I understand it, these entries exist only to trigger a change that get replicated over the topology:
  Their main purpose is to generate records in the changelog and they are updated from time to time by fractional
  replication  to insure that at least a change must be replicated by FR after a great number of not replicated changes are
  found in the changelog. The interest is that the fractional RUV get then updated so less changes need to be walked
    in the changelog when searching for the first change to send)

So creating a keep alive entry has little impact as far as the replication is fully started (including the changelog) 
    (As tbordaz mentioned, the main difference is that the master will appear in the RUV ))

IMHO the main interests of this fix are:
   - it is a good exercise for me to discover the 389 the open development process
   - more importantly, it will avoid having people getting puzzled by the fact that a master does not appear in the RUV 
     and spending time to understand why (I suspect that it what happened to lkrispenz and why he opened the issue -;) )

Reply:
That makes sense, and sounds good to me. I wonder if this (really good) explanation could be in the code comments somewhere? That way it's preserved for future-us to remember. Thank you!

Reply:
Hum, That seems a good idea: 
 the part about explaining the keep alive entries role could be set in comment just before the definition of the
  replica_subentry_* functions that creates, checks and updates the keep alive entries.